### ISSUE TITLE
Add Milestones or Roadmap

### ISSUE THREAD

**Issue by Heus-Sueh** [beeware/toga#2306]
### What is the problem or limitation you are having?

For now, the project does not have a Roadmap other than separate issues that could be organized in the Project tab on GitHub

### Describe the solution you'd like

I think it would be good to create a milestone or roadmap to track things that are missing or being worked on
It would also be good to organize issues related to specific aspects such as accessibility, APIs, Widgets...etc.

### Describe alternatives you've considered

None

### Additional context

Example:
![image](https://github.com/beeware/toga/assets/98482469/476d5fbd-5166-4567-ab71-ca50395da98c)

**Comment by freakboy3742** [comment]
Closing WONTFIX, on the basis that I've been down this path before, and it doesn't map well on to how open source projects work.

If you're a large engineering company with dedicated resources, you can make a plan, schedule work to be done, and signal which issues will be closed, on what timeframe. However an open source project doesn't work like that. We are at the mercy of the contributions we receive. Essentially the only person whose resources I can "allocate" are my own. 

As an example - a markup language is *very* low on my personal list of priorities (#2293). I'm unlikely to work on it myself. However, if an implementation landed in my inbox tomorrow, I'd review it and work with the contributor to get it merged. #2293 wouldn't appear on any milestone list... and yet it's just overridden the list of things that *are* on the milestone list.

The only point at which milestones become useful is when they become a list of blockers for the next release. However, we're not at the point of maturity where release blockers are a thing we need to manage. All bugs are bugs, and we tackle them as best we can.

Instead of a Github milestone list, we publish quarterly roadmaps (most recently, [Q3 2023](https://beeware.org/news/buzz/2023q4-roadmap/); our Q1 2024 roadmap will be published early in the new year). This gives a soft indication of where we will be focussing our effort over the coming 3 months, with a view to the long-term. Those documents start with a clear caveat - they're a guide to what *might* happen, not a hard commitment - specifically because we're not in a position to *make* hard commitments.

**Comment by mhsmith** [comment]
There actually is a [roadmap page](https://toga.readthedocs.io/en/v0.4.1/background/project/roadmap.html) in the documentation, but it doesn't really reflect our current priorities, so maybe it should be removed.

**Comment by Heus-Sueh** [comment]
> There actually is a [roadmap page](https://toga.readthedocs.io/en/latest/background/project/roadmap.html) in the documentation, but it doesn't really reflect our current priorities, so maybe it should be removed.

I had searched for "Roadmap" and didn't find anything, now I know why, you have to write "Road Map" separately

**Comment by Heus-Sueh** [comment]
> Closing WONTFIX, on the basis that I've been down this path before, and it doesn't map well on to how open source projects work.
> 
> If you're a large engineering company with dedicated resources, you can make a plan, schedule work to be done, and signal which issues will be closed, on what timeframe. However an open source project doesn't work like that. We are at the mercy of the contributions we receive. Essentially the only person whose resources I can "allocate" are my own.
> 
> As an example - a markup language is _very_ low on my personal list of priorities (#2293). I'm unlikely to work on it myself. However, if an implementation landed in my inbox tomorrow, I'd review it and work with the contributor to get it merged. #2293 wouldn't appear on any milestone list... and yet it's just overridden the list of things that _are_ on the milestone list.
> 
> The only point at which milestones become useful is when they become a list of blockers for the next release. However, we're not at the point of maturity where release blockers are a thing we need to manage. All bugs are bugs, and we tackle them as best we can.
> 
> Instead of a Github milestone list, we publish quarterly roadmaps (most recently, [Q3 2023](https://beeware.org/news/buzz/2023q4-roadmap/); our Q1 2024 roadmap will be published early in the new year). This gives a soft indication of where we will be focussing our effort over the coming 3 months, with a view to the long-term. Those documents start with a clear caveat - they're a guide to what _might_ happen, not a hard commitment - specifically because we're not in a position to _make_ hard commitments.

I understood, but maybe I expressed myself wrong, the idea would be to organize the issues and not give a deadline for when they would be corrected or implemented, especially because the open source world is unpredictable and it is impossible to give a deadline

**Comment by freakboy3742** [comment]
Yes - but what "organisation" would make sense? What meaningful criteria could we use to classify a ticket as being on the "next release"?

All bugs are candidates for being fixed in the next release. The bugs that get fixed are the ones that someone contributes a patch for. If we don't get a patch, the issue doesn't get closed. 

The publication of a release isn't gated on specific issues being addressed. We review and merge the PRs that people contribute. If those are bug fixes, then those bugs are the ones that are fixed. 

The publication of a release isn't gated on specific features being added. We cut a new release when there has been sufficient since the last release. "Sufficient" is a very loose definition - it usually comes down to "We're about to go to a conference and want to show off new features" or "we're getting tired of answering 'that's been fixed in main'" to certain problems.

So - a roadmap list of issues won't tell you what *will* be addressed, or *when* it will be addressed. What, then, would be the purpose of such a list?
